### PR TITLE
python3 compatible parse_variables

### PR DESCRIPTION
**Pull request by bqpd**

### CODE REVIEW

File: gpkit/tests/t_model.py [hunk 1]
```diff
@@ -740,15 +740,11 @@ def test_modelcontainmentprinting(self):
...
         # variables looked up from other models
-        if sys.version_info >= (3, 0):
-            with self.assertRaises(FutureWarning):
-                box = Box()
-        else:
-            box = Box()
-            area_bounds = BoxAreaBounds(box)
-            M = Model(box["V"], [box, area_bounds])
-            for var in ("h", "w", "d"):
-                self.assertEqual(len(M.variables_byname(var)), 1)
```
Comment (whoburg) [inline gpkit/tests/t_model.py]: Nice, good call getting rid of this

File: gpkit/tools/docstring.py [hunk 2]
```diff
@@ -57,17 +58,62 @@ def expected_unbounded(instance, doc):
...
+        second_indent = orig_lines[next_indented_idx][:indent_length]
+        parse_lines = [second_indent + line + "\n"
+                       for line in parse_varstring(self.string).split("\n")]
+        # make ast of these new lines, insert it into the original ast
+        new_lines = [orig_lines[1]] + parse_lines + orig_lines[2:]
+        new_src = "\n".join([l[first_indent_length:-1] for l in new_lines])
+        new_ast = ast.parse(new_src, "<parse_variables>")
+        ast.increment_lineno(new_ast, n=lineno-len(parse_lines))
+        code = compile(new_ast, inspect.getsourcefile(function), "exec",
+                       dont_inherit=True)  # don't inherit __future__ from here
```
Comment (whoburg) [inline gpkit/tools/docstring.py]: huh cool, learned something new

File: gpkit/tools/docstring.py [hunk 3]
```diff
@@ -57,17 +58,62 @@ def expected_unbounded(instance, doc):
...
-                                constant_declare)
-    out += check_and_parse_flag(string, "Variables\n", errorcatch)
-    out += check_and_parse_flag(string, "Variables of length", errorcatch,
-                                vv_declare)
-    return out
+    consts = check_and_parse_flag(string, "Constants\n", constant_declare)
+    variables = check_and_parse_flag(string, "Variables\n")
+    vecvars = check_and_parse_flag(string, "Variables of length", vv_declare)
+    out = ["# " + line for line in string.split("\n")]
+    out[0] = "from gpkit import Variable, VectorVariable" + "  " + out[0]
```
Comment (whoburg) [inline gpkit/tools/docstring.py]: inline comment for this? Will the two imports ever change / need to be expanded?

File: gpkit/tools/docstring.py [hunk 4]
```diff
@@ -112,14 +161,15 @@ def check_and_parse_flag(string, flag, errorcatch, declaration_func=None):
                 while line[labelstart] == " ":
                     labelstart += 1
                 label = line[labelstart:].replace("'", "\\'")
-            countstr += variable_declaration(nameval, units, label, line,
-                                             errorcatch)
+            countstr += variable_declaration(nameval, units, label, line)
+            # note that this is a 0-based line indexing
```
Comment (whoburg) [inline gpkit/tools/docstring.py]: 👍

File: pylint.sh [hunk 5]
```diff
@@ -12,4 +12,4 @@ export PYTHONPATH=$PYTHONPATH:$WORKSPACE/gpkit/
 
 python $PYLINT --rcfile=$WORKSPACE/.pylintrc --disable=fixme $@ $WORKSPACE/gpkit/
 
-python $PYLINT --rcfile=$WORKSPACE/.pylintrc --disable=superfluous-parens,exec-used,undefined-variable,no-member,not-callable,attribute-defined-outside-init,invalid-name,too-many-locals,redefined-outer-name,wrong-import-position $@ $WORKSPACE/docs/source/examples/*.py
+python $PYLINT --rcfile=$WORKSPACE/.pylintrc --disable=superfluous-parens,undefined-variable,no-member,not-callable,attribute-defined-outside-init,invalid-name,too-many-locals,redefined-outer-name,wrong-import-position $@ $WORKSPACE/docs/source/examples/*.py
```
Comment (whoburg) [inline pylint.sh]: Wonderful! Really happy to see this list getting shorter. Great SA to keep shrinking it.

File: gpkit/tools/docstring.py (hunk 3 above)
Comment (bqpd) [reply gpkit/tools/docstring.py]: not unless the sections just above are changed; added a comment!

**Comment by bqpd** [comment]
test models please

**Comment by bqpd** [comment]
test models please

**Comment by bqpd** [comment]
test models please

**Comment by bqpd** [comment]
test models please

**Pull request merged**